### PR TITLE
Docs/restructure and improve READMEs

### PR DESCRIPTION
**Pull request by plumdumpling**
- split library README into separate files
- add README to examples directory
- use relative paths in GitHub README
- improve library documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,20 @@
 # Google Maps React Hooks
-[![npm version](https://img.shields.io/npm/v/@ubilabs/google-maps-react-hooks)](https://www.npmjs.com/package/@ubilabs/google-maps-react-hooks) [![GitHub license](https://img.shields.io/badge/license-MIT-green.svg)](https://github.com/ubilabs/google-maps-react-hooks/blob/develop/LICENSE)
+[![npm version](https://img.shields.io/npm/v/@ubilabs/google-maps-react-hooks)](https://www.npmjs.com/package/@ubilabs/google-maps-react-hooks) [![GitHub license](https://img.shields.io/badge/license-MIT-green.svg)](./LICENSE)
 
 ## Description
 
 This is a JavaScript library to easily implement a Google Maps map into your React application. It comes with a collection of React hooks to access the Google Maps map instance all over your components and to handle some common interactions with the map.
```
Comment (BiniCodes) [inline README.md:6]: Maybe we can add/change the last sentence to sth. like:

```suggestion
This is a JavaScript library to easily implement a Google Maps map into your React application. It comes with a collection of React hooks to access the Google Maps map instance all over different components. It also provides different hooks to use [Google Map services](https://developers.google.com/maps/documentation/javascript/services) and handle common map interactions.
```

File: README.md [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
-- [Google Maps React Hooks Library](https://github.com/ubilabs/google-maps-react-hooks/blob/main/library/README.md)
-- [Examples](https://github.com/ubilabs/google-maps-react-hooks/tree/main/examples)
-  - [Basic Google Map](https://github.com/ubilabs/google-maps-react-hooks/blob/main/examples/basic-google-map)
-  - [Google Map with Markers](https://github.com/ubilabs/google-maps-react-hooks/blob/main/examples/google-map-with-markers)
-  - [Places Autocomplete](https://github.com/ubilabs/google-maps-react-hooks/blob/main/examples/places-autocomplete)
-  - [Directions Service](https://github.com/ubilabs/google-maps-react-hooks/blob/main/examples/directions)
-  - [Geocoding Service](https://github.com/ubilabs/google-maps-react-hooks/blob/main/examples/geocoding)
-  - [Places Library](https://github.com/ubilabs/google-maps-react-hooks/blob/main/examples/places)
-  - [Distance Matrix Service](https://github.com/ubilabs/google-maps-react-hooks/blob/main/examples/distance-matrix)
+- [Google Maps React Hooks Library](./library)
```
Comment (BiniCodes) [inline README.md:9]: This not linking to the `README.md` but to the director itself.

```suggestion
- [Google Maps React Hooks Library](./library/README.md)
```

Same for all the links below...

File: README.md [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+- [Examples](./examples)
```
Comment (BiniCodes) [inline README.md:10]:
```suggestion
- [Examples](./examples/README.md))
```

File: README.md [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
+  - [Basic Google Map](./examples/basic-google-map)
```
Comment (BiniCodes) [inline README.md:11]:
```suggestion
  - [Basic Google Map](./examples/basic-google-map/README.md)
```

and so on for the other one belows :)

File: examples/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Google Maps React Hooks Examples
+
+Here you can find some implementation examples for the [Google Maps React Hooks library](../library).
+
+- [Basic Google Map](./basic-google-map)
```
Comment (BiniCodes) [inline examples/README.md:5]: Should be README again here, or?

```suggestion
- [Basic Google Map](./basic-google-map/README.md)

(Same for below, if you did not want to link to the directory itsef)
```

File: README.md (hunk 2 above)
Comment (plumdumpling) [reply README.md:9]: This was intended. If you open a directory that includes a README file, GitHub displays the content of the README right below the content of the directory.

File: README.md (hunk 3 above)
Comment (plumdumpling) [reply README.md:10]: See comment above

File: README.md (hunk 4 above)
Comment (plumdumpling) [reply README.md:11]: See comment above

File: examples/README.md (hunk 5 above)
Comment (plumdumpling) [reply examples/README.md:5]: See comment above

File: README.md (hunk 2 above)
Comment (BiniCodes) [reply README.md:9]: Okay, thought it should just navigate between the READMEs. But then all good

**Review by BiniCodes — APPROVED** [review]

**Pull request merged**